syntax = "proto3";

package calculateservice;

service CalculateService {
    rpc Calculate (Input) returns (Output) {}
}

message Input {
    double a0 = 1;
    double a1 = 2;
    double a2 = 3;
    double a3 = 4;
    double u = 5;
    double X0_0 = 6;
    double X0_1 = 7;
    double X0_2 = 8;
}

message Output {
    double X_0 = 1;
    double X_1 = 2;
    double X_2 = 3;
    double X_next_0 = 4;
    double X_next_1 = 5;
    double X_next_2 = 6;
}